components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1539,21 +1402,31 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           !          cannot possibly overdeplete qv
 
           ! cloud
-          sinks   = (qcaut+qcacc+qccol+qcevp+qcheti+qcshd)*dt
+          sinks   = (qcaut+qcacc+qccol+qcheti+qcshd+berg)*dt  !PMC remove ssat (added berg)
           sources = qc(i,k) + (qccon+qcnuc)*dt
           if (sinks.gt.sources .and. sinks.ge.1.e-20_rtype) then
              ratio  = sources/sinks
              qcaut  = qcaut*ratio
              qcacc  = qcacc*ratio
-             qcevp  = qcevp*ratio
              qccol  = qccol*ratio
              qcheti = qcheti*ratio
              qcshd  = qcshd*ratio
+ !+++PMC remove ssat
+             berg   = berg*ratio
+
+             !PMC 12/3/12: ratio is also frac of step w/ liquid.
+             !thus we apply berg for "ratio" of timestep and vapor
+             !deposition for the remaining frac of the timestep. 
+             qidep = qidep*(1._rtype-ratio) ! AaronDonahue: Is there any chance this could lead to unrealistic value since qidep isn't used to determine ratio in this case?
```

Comment:
I think you're saying that qidep would increase cloud mass, so removal ratio "ratio" wouldn't be accurate any more. That's not the case because qidep takes vapor and turns it into ice, so it doesn't affect the amount of qc available.

Reply:
Ok great, just wanted to make sure we gave this term a second look.